### PR TITLE
Switch to out-source builds for LibTorch

### PR DESCRIPTION
**Pull request by peterjc123**

### CODE REVIEW

**Review by yf225 — APPROVED** [review]

File: .circleci/scripts/binary_checkout.sh [hunk 1]
```diff
@@ -39,10 +39,10 @@ git --no-pager log --max-count 1
 popd
 
 # Clone the Builder master repo
-git clone -q https://github.com/pytorch/builder.git "$BUILDER_ROOT"
+git clone -q https://github.com/peterjc123/builder.git "$BUILDER_ROOT"
 pushd "$BUILDER_ROOT"
 git fetch origin
-git reset origin/master --hard
+git reset origin/path_fix --hard
```
Comment (yf225) [inline .circleci/scripts/binary_checkout.sh]: We can merge the PR after removing the changes to this file.

Comment (peterjc123) [reply .circleci/scripts/binary_checkout.sh]: Sure. I'll revert it back when the whole test matrix is completed.

Comment (peterjc123) [reply .circleci/scripts/binary_checkout.sh]: Reverted!

**Comment by ezyang** [comment]
CI failure is real:

```

Jun 14 17:42:10 /pytorch ~/project
Jun 14 17:42:10 ++ mkdir -p libtorch/lib libtorch/bin libtorch/include libtorch/share
Jun 14 17:42:10 ++ cp -r build/lib libtorch/
Jun 14 17:42:10 cp: cannot stat ‘build/lib’: No such file or directory
```

**Comment by yf225** [comment]
@ezyang CI failure is fixed after https://github.com/pytorch/builder/pull/314 is merged. I can rerun the `binary_linux_libtorch_2.7m_cpu_devtoolset3_build` job to make sure it works now.

**Pull request closed without merging**